src/databricks/labs/ucx/framework/owners.py

```diff
@@ -0,0 +1,204 @@
+        """Determine whether a user has a given role or not."""
+        return user.roles is not None and any(r.value == role for r in user.roles)
+
+    def find_admin_users(self) -> Iterable[User]:
+        """Enumerate the active account administrators associated with a given workspace.
+
+        Returns:
+            Iterable[User]: The active account administrators, if any.
+        """
+        logger.debug("Enumerating account users to locate active administrators...")
+        response = self._ws.api_client.do(
+            "GET", "/api/2.0/account/scim/v2/Users", query={"attributes": "id,active,userName,roles"}
+        )
+        assert isinstance(response, dict)
+        all_users = (User.from_dict(resource) for resource in response.get("Resources", []))
+        # Reference: https://learn.microsoft.com/en-us/azure/databricks/admin/users-groups/groups#account-admin
+        return (user for user in all_users if user.active and user.user_name and self._has_role(user, "account_admin"))
+
+
+class AdministratorLocator:
+    """Locate a workspace administrator, if possible.
+
+    This will first try to find an active workspace administrator. If there are multiple, the first (alphabetically
+    sorted by user-name) will be used. If no active workspace administrators can be found then an account administrator
+    is sought, again returning the first alphabetically by user-name if more than one is found.
+    """
+
+    def __init__(
+        self,
+        ws: WorkspaceClient,
+        *,
+        finders: Sequence[Callable[[WorkspaceClient], AdministratorFinder]] = (
+            WorkspaceAdministratorFinder,
+            AccountAdministratorFinder,
+        ),
+    ) -> None:
+        """
+        Initialize the instance, which will try to locate administrators using the workspace for the supplied client.
+
+        Args:
+            ws (WorkspaceClient): the client for workspace in which to locate admin users.
+            finders: a sequence of factories that will be instantiated on demand to locate admin users.
+        """
+        self._ws = ws
+        self._finders = finders
+
+    @cached_property
+    def _workspace_id(self) -> int:
+        # Makes a REST call, so we cache it.
+        return self._ws.get_workspace_id()
+
+    @cached_property
+    def _found_admin(self) -> str | None:
+        # Lazily instantiate and query the finders in an attempt to locate an admin user.
+        finders = (finder(self._ws) for finder in self._finders)
+        # If a finder returns multiple admin users, use the first (alphabetically by user-name).
+        first_user = functools.partial(min, default=None, key=lambda user: user.user_name)
+        found_admin_users: Iterable[User | None] = (first_user(finder.find_admin_users()) for finder in finders)
+        return next((user.user_name for user in found_admin_users if user), None)
+
+    @property
+    def workspace_administrator(self) -> str:
```

Comment:
```suggestion
    def get_workspace_administrator(self) -> str:
```
let's not make public `@property` with side-effects, like calling APIs.